### ISSUE TITLE
Support checking python file without extension

### ISSUE THREAD

**Issue by bulletmark** [astral-sh/ty#1692]
### Summary

`ty` will not check a file unless it has a python extension, e.g:

```sh
$ ty check -v myscript 
INFO Defaulting to python-platform `linux`
INFO Python version: Python 3.13, platform: linux
INFO Indexed 0 file(s) in 0.003s
WARN No python files found under the given path(s)
All checks passed!
```

Whereas:

```sh
$ ruff check -v myscript 
[2025-12-01][09:48:38][ruff::resolve][DEBUG] Using configuration file (via parent) at: /home/mark/Data/comps/aoc/pyproject.toml
[2025-12-01][09:48:38][ignore::gitignore][DEBUG] opened gitignore file: /home/mark/.config/git/ignore
[2025-12-01][09:48:38][ruff::commands::check][DEBUG] Identified files to lint in: 3.846741ms
[2025-12-01][09:48:38][ruff::diagnostics][DEBUG] Checking: /home/mark/Data/comps/aoc/2024/myscript
[2025-12-01][09:48:38][ruff::commands::check][DEBUG] Checked 1 files in: 785.466µs
All checks passed!
```

### Version

ty 0.0.1-alpha.29

**Comment by sharkdp** [comment]
I also wanted this feature several times. It would also give us the option to type-check code that's being produced by another process, even without adding a `-` special argument:
```bash
ty check <(echo "reveal_type(1)")
```

**Comment by Gankra** [comment]
Micha introduced `System::source_type` to handle a similar problem for notebooks. Sounds like we should be using it in more places.

**Comment by refi64** [comment]
It's worth noting that this *also affects the LSP*. e.g. if I have this in both `x` and `x.py`:

```python
#!/usr/bin/env python3
print("s" + 1)
```

then opening `x` shows no errors, but `x.py` works as expected:

<img width="1718" height="528" alt="Image" src="https://github.com/user-attachments/assets/d65ee2ce-5fff-4861-94f6-3f237777c2e8" />

That being said, operations like hover and autocomplete work in *both* files; it's only the checks that don't work.
